apple/internal/utils/main_thread_checker_dylibs.bzl

```
@@ -0,0 +1,40 @@
+# Copyright 2022 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Support functions related to getting main thread checker libraries."""
+
+load("@bazel_tools//tools/cpp:toolchain_utils.bzl", "find_cpp_toolchain")
+
+def _should_package_main_thread_checker_dylib(*, features):
+    """Returns whether the libMainThreadChecker.dylib should be bundled."""
+
+    return "include_main_thread_checker" in features
```

Other rules_apple features have an `apple.` prefix. Also I suggest calling this `main_thread_checker` just like the same feature name in Xcode.
```suggestion
    return "apple.main_thread_checker" in features
```